### PR TITLE
update development doc and regenerate clustermanagementaddons crd

### PR DESCRIPTION
**Pull request by hanqiuzh**
Updates:
- added steps for generating CRDs & generating deepcopy.zz
- updated docker image to use `fedora:32` to support golang 1.14.4+
- updated clustermanagementaddon's printcolumn by regenerating

### CODE REVIEW

**Comment by hanqiuzh** [comment]
@qiujian16  would you like to check the docs, and please let me know if it align with what you've done for generating crds? Thanks

File: Dockerfile.build [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM fedora:31
+FROM fedora:32
```
Comment (pmorie) [inline Dockerfile.build:1]: why does this need to change in this PR?

File: docs/development.md [hunk 2]
```diff
@@ -34,5 +49,15 @@ After this, calling `make update-codegen-crds` should generate a new structural
 - Ensure that your API is correctly declared for the generator to pick it up. That means, in your `doc.go`, include the following:
   1. `// +groupName=<API_GROUP_NAME>`, this should match the `group` in your CRD `spec`
   2. `// +kubebuilder:validation:Optional`, this tells the operator that fields should be optional unless explicitly marked with `// +kubebuilder:validation:Required`
+- This will not touch any non-schema fields in the CRD. For example, `additionalPrinterColumns` & `preserveUnknownFields`. To regenerate those fields, please follow the instructions in the [Generating CRD for the first time](#Generating-CRD-for-the-first-time) section.
 
-For more information on the API markers to add to your Go types, see the [Kubebuilder book](https://book.kubebuilder.io/reference/markers.html)
\ No newline at end of file
+For more information on the API markers to add to your Go types, see the [Kubebuilder book](https://book.kubebuilder.io/reference/markers.html)
+
+## Generating scripts
```
Comment (pmorie) [inline docs/development.md]: s/Generating scripts/Generating code/

File: docs/development.md [hunk 3]
```diff
@@ -34,5 +49,15 @@ After this, calling `make update-codegen-crds` should generate a new structural
 - Ensure that your API is correctly declared for the generator to pick it up. That means, in your `doc.go`, include the following:
   1. `// +groupName=<API_GROUP_NAME>`, this should match the `group` in your CRD `spec`
   2. `// +kubebuilder:validation:Optional`, this tells the operator that fields should be optional unless explicitly marked with `// +kubebuilder:validation:Required`
+- This will not touch any non-schema fields in the CRD. For example, `additionalPrinterColumns` & `preserveUnknownFields`. To regenerate those fields, please follow the instructions in the [Generating CRD for the first time](#Generating-CRD-for-the-first-time) section.
```
Comment (pmorie) [inline docs/development.md:52]: Thanks for adding this note :)

File: Dockerfile.build (hunk 1 above)
Comment (hanqiuzh) [reply Dockerfile.build:1]: golang.mk make commands requires golang 1.14.4+. `fedora:31` can only install golang 1.13, will have the following error:
```
vendor/github.com/openshift/build-machinery-go/make/targets/golang/../../lib/golang.mk:22: *** `go` is required with minimal version "1.14.4", detected version "1.13.15". You can override this check by using `make GO_REQUIRED_MIN_VERSION:=`.  Stop.
```

I tried to use `dnf -y update`, but still installed golang 1.13, so I used `fedora:32` instead (which install golang 1.15).

Comment (pmorie) [reply Dockerfile.build:1]: I see, thanks

**Comment by pmorie** [comment]
/approve

on @qiujian16 for lgtm

**Comment by qiujian16** [comment]
/lgtm

thanks! @hanqiuzh

**Pull request merged**